File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-18.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0, r = 0;
        int res = 0;
        unordered_map<char, int> mp;
        while( r < s.size()){
            if(mp.count(s[r])){
                l = max(mp[s[r]]+1, l);
            }
            mp[s[r]] = r;
            res = max(res, r-l+1);
            r++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-18.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l , r = 1, max(piles)
        while l < r:
            mid = (l+r) //2
            res = 0
            for i in piles:
                if i <= mid:
                    res +=1
                    continue
                res += (i + mid -1 ) // mid
            if res <= h:
                r = mid
            else:
                l = mid+1
        return l
                


